templates/IAM/oidc-provider.yaml

```diff
@@ -0,0 +1,29 @@
+AWSTemplateFormatVersion: 2010-09-09
+Description: Setup an OIDC provider
+Parameters:
+  ClientIdList:
+    Type: List<String>
+    Description: >-
+      A list of client IDs (also known as audiences) that are associated with
+      the specified IAM OIDC provider resource object
+    Default: "sts.amazonaws.com"
+  ThumbprintList:
+    Type: List<String>
+    Description: >-
+      A list of certificate thumbprints that are associated with the specified
+      IAM OIDC provider resource object
+  Url:
+    Type: String
+    Description: "The URL that the IAM OIDC provider resource object is associated with"
+Resources:
+  Provider:
+    Type: AWS::IAM::OIDCProvider
```

Comment:
why do we need this template at all?  

also, instead of creating this wrapper why would you not just use this CFN resources directly?